src/hardware_functions.c

```diff
@@ -1,4 +1,8 @@
 #include "hardware_functions.h"
+#include "stdio.h"
```

Comment:
hoort deze niet binnen een ander file te staan? (of staat ie niet al in hardware functions?